File: mytext.ts
import {HTMLBox, HTMLBoxView} from "models/layouts/html_box"

import {div, span} from "core/dom"
import * as p from "core/properties"

export class MyTextView extends HTMLBoxView {
  model: MyText

  render(): void {
    super.render()
    const m = this.model
    const sentence = m.text.map((word, i) => span({style: `color: ${m.color[i]}; font-size: ${m.size[i]}px;`}, `${word} `))
    this.el.appendChild(div({}, sentence))
  }
}

export namespace MyText {
  export type Attrs = p.AttrsOf<Props>
  export type Props = HTMLBox.Props & {
    text: p.Property<string[]>
    color: p.Property<string[]>
    size: p.Property<number[]>
  }
}

export interface MyText extends MyText.Attrs {}

export class MyText extends HTMLBox {
  properties: MyText.Props

  static init_MyText(): void {
    this.prototype.default_view = MyTextView
    this.define<MyText.Props>({
      text: [p.Array, []],
      color: [p.Array, []],
      size: [p.Array, []],
    })
  }
}

File: mytext.py
#!/usr/bin/env python3

from bokeh.core.properties import Int, String, List
from bokeh.layouts import column
from bokeh.models.layouts import Column, HTMLBox
from bokeh.io import show

class MyText(HTMLBox):
    __implementation__ = 'mytext.ts'
    text = List(String)
    color = List(String)
    size = List(Int)

red, green, blue = '#ff0000', '#00ff00', '#0000ff'

show(column(MyText(text=['This', 'is', 'a', 'pen.'],
                   color=[red, green, blue, red],
                   size=[12, 24, 36, 48])))
